tools/challenge-md-parser/fix-challenges.js

```diff
@@ -0,0 +1,397 @@
+
+  const file = await vfile.read(filePath);
+  file.data = {
+    ...file.data,
+    lang,
+    referenceChallenge,
+    translateText
+  };
+  return await processor.process(file);
+}
+
+async function getReferenceChallengeData(filePath) {
+  const parts = filePath.split(path.sep);
+  parts.push(parts.pop().replace(/\.[^.]+\.md$/, '.english.md'));
+  parts[parts.length - 4] = 'english';
+  const filePathEnglishChallenge = parts.join(path.sep);
+  try {
+    const fileData = await vfile.read(filePathEnglishChallenge);
+    fileData.data = { ...fileData.data, refData: true };
+    return (await processor.process(fileData)).data;
+  } catch (err) {
+    console.error(err);
+    return null;
+  }
+}
+
+function detectLang(filePath) {
+  const match = /\.([^.]+)\.md$/.exec(filePath);
+  if (!match) {
+    throw new Error(`Incorrect file path ${filePath}`);
+  }
+  return match[1];
+}
+
+function frontmatterToData() {
+  return transformer;
+
+  function transformer(tree, file) {
+    visit(tree, 'yaml', visitor);
+
+    function visitor(node) {
+      const frontmatter = node.value;
+
+      file.data = { ...file.data, frontmatter };
+    }
+  }
+}
+
+function testsToData() {
+  return (tree, file) => {
+    visit(tree, 'code', visitor);
+
+    function visitor(node) {
+      const { lang, value } = node;
+      if (lang === 'yml') {
+        file.data = {
+          ...file.data,
+          tests: value
+        };
+      }
+    }
+  };
+}
+
+function textToData(sectionIds) {
+  return (tree, file) => {
+    let indexId = 0;
+    let currentSection = sectionIds[indexId];
+    let inSection = false;
+    let nodes = [];
+    let findSection;
+    const visitor = (node, index, parent) => {
+      if (!parent) {
+        return visit.CONTINUE;
+      }
+
+      if (node.type === 'heading') {
+        if (inSection) {
+          findSection = new RegExp(`^<section id=('|")${currentSection}\\1>`);
+          file.data = {
+            ...file.data,
+            [currentSection]: new stringify.Compiler(
+              { type: 'root', children: nodes },
+              file
+            )
+              .compile()
+              .trim()
+              .replace(findSection, '')
+              .replace(/<\/section>$/, '')
+              .trim()
+          };
+          nodes = [];
+          indexId++;
+          if (indexId < sectionIds.length) {
+            currentSection = sectionIds[indexId];
+          } else {
+            return visit.EXIT;
+          }
+        }
+        inSection = true;
+      } else if (inSection) {
+        nodes.push(node);
+      }
+
+      return visit.SKIP;
+    };
+    visit(tree, visitor);
+  };
+}
+
+function createTranslateText(target, source = 'english') {
+  const projectId = process.env.GOOGLE_CLOUD_PROJECT_ID;
+  if (!projectId) {
+    return async text => text;
+  }
+  const languageCodes = {
+    arabic: 'ar',
+    chinese: 'zh',
+    english: 'en',
+    portuguese: 'pt',
+    russian: 'ru',
+    spanish: 'es'
```

Comment:
I'm wondering if this syntax is good enough. I'm sure it won't pick Catalan (but I also remember the translators prefering LATAM Spanish), but for Portuguese (_PT, _BR) and Chinese (traditional or simplified), it's less clear. It's our choice to make.